### PR TITLE
Add shared test coverage guardrail base

### PR DESCRIPTION
**Pull request by LeonardoRosaa**
## What has changed and why?

Introduces `coverage-base.ts`, a generic factory (`createCoverageGuardrail`) that encapsulates the full coverage-check pipeline in a single reusable function. Concrete guardrails (e.g. a future TypeScript test-coverage guardrail) supply a `CoverageConfig` with four functions:           
                                                                                              
  - filterFiles - select relevant source files from the PR diff
  - findTestFile - resolve the test file for a given source path
  - runTests - execute the test suite and return raw coverage data
  - parseCoverageRatio - extract the line-coverage ratio for added lines                        

The base handles the rest: skipping deleted files and files without a patch, batching all source files into a single test run, deduplicating test files, enforcing the 80% threshold on newly added lines only, and producing consistent [PASS]/[FAIL] output.

## How has it been tested?

Unit tests.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Added a coverage-based guardrail that checks changed source files against related tests and reports pass/fail results.
  * Improved handling for edge cases such as missing test coverage data, deleted files, and files without patches.
  * Added more detailed coverage summaries, including percentage-based results and combined failure reporting.

### CODE REVIEW

File: fast_track/src/guardrails/shared/utils.ts [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+
+        if (line.startsWith('+')) {
+            added.add(newLine);
+            newLine++;
+        } else if (line.startsWith(' ')) {
+            // Context line — present in new file but not added.
+            newLine++;
+        }
+        // Lines starting with '-' do not appear in the new file; newLine stays put.
+    }
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply fast_track/src/guardrails/shared/utils.ts:32]: Done

File: fast_track/src/guardrails/shared/coverage-base.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        async run(ctx: GuardrailContext): Promise<GuardrailResult> {
+            const files = await ctx.changedFiles();
+            const sourceFiles = config.filterFiles(files).filter((f) => f.status !== 'deleted');
+
+            if (sourceFiles.length === 0) {
+                return { name: config.name, status: 'pass', summary: '0 file(s) checked.' };
+            }
+
+            const addedLinesByFile = buildAddedLinesMap(sourceFiles);
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply fast_track/src/guardrails/shared/coverage-base.ts:108]: Done

File: fast_track/src/guardrails/shared/coverage-base.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+const MIN_COVERAGE = 0.8;
+const MIN_COVERAGE_PCT = `${(MIN_COVERAGE * 100).toFixed(0)}%`;
+
+interface CoverageConfig<TCoverage> {
+    name: string;
+    filterFiles(files: ChangedFile[]): ChangedFile[];
+    findTestFile(sourcePath: string): Promise<string | undefined>;
+    runTests(testFiles: string[], sourcePaths: string[]): Promise<TCoverage | null>;
+    parseCoverageRatio(data: TCoverage, sourcePath: string, addedLines: Set<number>): number | null;
+}
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply fast_track/src/guardrails/shared/coverage-base.ts]: Done

**Review by ikondrat — APPROVED** [review]
